### ISSUE TITLE
In for loop a semicolon in not checked for existence

### ISSUE THREAD

**Issue by dandanio** [koalaman/shellcheck#1273]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

for drive in /dev/sd[a-z]
do
    echo "${drive}"
    smartctl -a "${drive}" | grep -E '(Reallocated_Sector|Pending_Sector|Offline_Uncorrectable)'
    echo
done
```

#### Here's what shellcheck currently says:
```

```
#### Here's what I wanted or expected to see:
"; expected" or something similar with a for loop example

Bash shows:
**Syntax error: word unexpected (expecting "do")**

A working example is:
```sh
#!/usr/bin/env bash

for drive in /dev/sd[a-z];
do
    echo "${drive}"
    smartctl -a "${drive}" | grep -E '(Reallocated_Sector|Pending_Sector|Offline_Uncorrectable)'
    echo
done
```

**Comment by ngzhian** [comment]
It works with or without ;, what version of bash are you using?

**Comment by koalaman** [comment]
I'm unable to reproduce this. Please reopen with additional information.
